src/coordination/coordinator_data.cpp

```diff
@@ -111,39 +112,43 @@ auto CoordinatorData::DoFailover() -> DoFailoverStatus {
     return DoFailoverStatus::RPC_FAILED;
   }
 
+  auto old_main = std::ranges::find_if(registered_instances_, &CoordinatorInstance::IsMain);
+  // TODO: (andi) For performing restoration we will have to improve this
+  old_main->client_.PauseFrequentCheck();
+
   chosen_replica_instance->PostFailover(main_succ_cb_, main_fail_cb_);
 
   return DoFailoverStatus::SUCCESS;
 }
 
-auto CoordinatorData::ShowMain() const -> std::optional<CoordinatorInstanceStatus> {
-  std::shared_lock<utils::RWLock> lock{coord_data_lock_};
-  auto main_instance = std::ranges::find_if(registered_instances_, &CoordinatorInstance::IsMain);
-  if (main_instance == registered_instances_.end()) {
-    return std::nullopt;
-  }
+auto CoordinatorData::ShowInstances() const -> std::vector<CoordinatorInstanceStatus> {
+  std::vector<CoordinatorInstanceStatus> instances_status;
+  instances_status.reserve(registered_instances_.size());
 
-  return CoordinatorInstanceStatus{.instance_name = main_instance->InstanceName(),
-                                   .socket_address = main_instance->SocketAddress(),
-                                   .is_alive = main_instance->IsAlive()};
-};
+  auto const stringify_repl_role = [](const CoordinatorInstance &instance) -> std::string {
+    if (!instance.IsAlive()) return "";
```

Comment:
Maybe return something "na", "unknown" or similar.

Reply:
Will do it here https://github.com/memgraph/memgraph/pull/1671